### DISCUSSION TITLE
How do we investigate memory leak with Salt

### DISCUSSION THREAD

**Discussion by senthilkumar-e** [saltstack/salt#66226] [Q&A]
We are seeing memory leak with MworkerQueue process when some bad minion sends auth flood. Is there any tools or troubleshooting docs available to identify the code path which is leaking memory ?

**Comment by dwoz** [comment]
What version of Salt are you using?

**Comment by senthilkumar-e** [comment]
> What version of Salt are you using?

Salt Version:
          Salt: 3004.2
 
Dependency Versions:
          cffi: 1.15.1
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
      M2Crypto: 0.38.0
          Mako: Not Installed
       msgpack: 1.0.4
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.11.0
        pygit2: Not Installed
        Python: 3.9.5 (default, May  4 2023, 22:47:07)
  python-gnupg: Not Installed
        PyYAML: 6.0
         PyZMQ: 21.0.2
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.3
 
System Versions:
          dist: mariner 2.0 
        locale: utf-8
       machine: x86_64
       release: 5.15.111.1-1.cm2
        system: Linux
       version: Common Base Linux Mariner 2.0

**Comment by dwoz** [comment]
`3004.2` has been out of support for some time now. There have been several memory leak fixes in newer releases. Please upgrade to `3006.x`.

**Reply by tjyang** [reply]
@dwoz 
The memory leak symptom still exist up to 3007. This is a valid question.
 memory leak symptom = restart 3007 salt-master (without syndicate enabled) after a day or two of usage. 
What I like to know is how to display network port resource status of 4505 and 4506.
See My issue at  https://github.com/saltstack/salt/issues/66282

**Reply by tjyang** [reply]
See my notes here for memory leak tracking, https://github.com/saltstack/salt/issues/66282#issuecomment-2077258318
